packages/iris-grid/src/mousehandlers/IrisGridTokenMouseHandler.tsx

```diff
@@ -0,0 +1,133 @@
+import {
+  type EventHandlerResult,
+  type Grid,
+  type GridPoint,
+  isLinkToken,
+  GridTokenMouseHandler,
+  type GridRangeIndex,
+} from '@deephaven/grid';
+import deepEqual from 'fast-deep-equal';
+import type IrisGrid from '../IrisGrid';
+
+// Handler also helps with other tooltips
+class IrisGridTokenMouseHandler extends GridTokenMouseHandler {
+  private irisGrid: IrisGrid;
+
+  private lastColumn: GridRangeIndex;
+
+  private lastRow: GridRangeIndex;
+
+  constructor(irisGrid: IrisGrid) {
+    super();
+
+    this.irisGrid = irisGrid;
+    this.lastColumn = null;
+    this.lastRow = null;
+  }
+
+  private destroyTooltip(): void {
+    this.irisGrid.setState({ hoverTooltipProps: null });
+  }
+
+  protected setCursor(gridPoint: GridPoint, grid: Grid): EventHandlerResult {
+    if (this.isHoveringLink(gridPoint, grid)) {
+      this.cursor = 'pointer';
+      return { stopPropagation: false, preventDefault: false };
+    }
+    this.cursor = null;
+    return false;
+  }
+
+  onMove(gridPoint: GridPoint, grid: Grid): EventHandlerResult {
+    const { model } = this.irisGrid.props;
+    const isUserHoveringLink = this.isHoveringLink(gridPoint, grid);
+    const tooltip = model.tooltipForCell(gridPoint.column, gridPoint.row);
+
+    if (
+      isUserHoveringLink &&
+      this.currentLinkBox != null &&
+      isLinkToken(this.currentLinkBox.token)
+    ) {
+      const { hoverTooltipProps } = this.irisGrid.state;
+      if (this.currentLinkBox == null) {
+        return false;
+      }
+      const { x1: left, y1: top, x2: right, y2: bottom } = this.currentLinkBox;
+      const { href } = this.currentLinkBox.token;
+      const width = right - left;
+      const height = bottom - top;
+      const newProps = { left, top: top + 1, width, height };
+      if (!deepEqual(hoverTooltipProps, newProps)) {
```

Comment:
All of the variables seem to only be used if `!deepEqual(hoverTooltipProps, newProps)` is true. The same seems true for the `else if (tooltip !== null)` block. I would suggest checking `deepEqual(hoverTooltipProps, newProps)` before all of this logic and bailing early if it is true. This should cut down on some nested if / else blocks.

Reply:
```typescript
if (deepEqual(hoverTooltipProps, newProps)) {
  this.destroyTooltip();
}
else if (isUserHoveringLink &&
      this.currentLinkBox != null &&
      isLinkToken(this.currentLinkBox.token)) {
      ...
}
else if (tooltip !== null) {
   ...
}
```

Reply:
`newProps` is different depending on whether it's the link or tooltip condition. Would it still be ideal to have some logic like:
```javascript
let tooltip = {};
if (
  isUserHoveringLink &&
  this.currentLinkBox != null &&
   isLinkToken(this.currentLinkBox.token)
) {
  tooltip = { ... };
} else if (tooltip !== null) {
  tooltip = { ... };
}
```

or would it be better to just do:
```javascript
if (deepEqual(hoverTooltipProps, newProps)) continue;
```

Reply:
Ok yeah I missed that before.